### ISSUE TITLE
Loss funtions

### ISSUE THREAD

**Issue by JavierClearImageAI** [matterport/Mask_RCNN#1800]
At lines 1076 - 1109 of model.py:

 . . .
```
    # Loss
    loss = tf.nn.sparse_softmax_cross_entropy_with_logits(
        labels=target_class_ids, logits=pred_class_logits)

    # Erase losses of predictions of classes that are not in the active
    # classes of the image.
    loss = loss * pred_active

    # Computer loss mean. Use only predictions that contribute
    # to the loss to get a correct mean.
    loss = tf.reduce_sum(loss) / tf.reduce_sum(pred_active)
    return loss
```

I'm not sure to get it right. I see that you are using a variation of the standard cross-entropy loss function, but it doesn't seem to be the Focal Loss. Is there a reason for that? Could you explain in a couple of lines which loss function have you used? 

[Review: RetinaNet — Focal Loss (Object Detection)](https://towardsdatascience.com/review-retinanet-focal-loss-object-detection-38fba6afabe4)
[RetinaNet: how Focal Loss fixes Single-Shot Detection](https://towardsdatascience.com/retinanet-how-focal-loss-fixes-single-shot-detection-cb320e3bb0de).
[Focal Loss for Dense Object Detection - Facebook AI Research](http://openaccess.thecvf.com/content_ICCV_2017/papers/Lin_Focal_Loss_for_ICCV_2017_paper.pdf)
[mkocabas/focal-loss-keras](https://github.com/mkocabas/focal-loss-keras)

Thanks for the nice work!
